sdk/core/core-client-rest/src/getClient.ts

```diff
@@ -17,23 +17,23 @@ import { buildRequestUrl } from "./urlHelpers";
 
 /**
  * Creates a client with a default pipeline
- * @param baseUrl - Base endpoint for the client
+ * @param endpoint - Base endpoint for the client
  * @param options - Client options
  */
-export function getClient(baseUrl: string, options?: ClientOptions): Client;
+export function getClient(endpoint: string, options?: ClientOptions): Client;
```

Comment:
It seems like currently we didn't refer to the baseUrl in ClientOptions at all, I wonder if we pass both endpoint and options.endpoint, should we consider use options.endpoint in the core side ? Or we should just rely on the client side to pass the correct value to endpoint like 
```ts
  const endpoint =
    options.endpoint ?? options.baseUrl ??  `http://localhost:3000/client/structure/${clientParam}`;
  // other logic code
  const client = getClient(endpoint, options) as ServiceClient;
```

Reply:
Does this mean that even if customers passes options.baseUrl, it wouldn't take any effect?

I think your proposal makes sense, I wonder if we should log a warning if options.baseUrl is passed